invokeai/backend/stable_diffusion/diffusers_pipeline.py

```diff
@@ -457,6 +466,7 @@ def image_from_embeddings(
         noise: torch.Tensor,
         callback: Callable[[PipelineIntermediateState], None] = None,
         run_id=None,
+        **kwargs,
```

Comment:
rather than propagating `kwargs` can you put the extra conditioning info into the `ExtraConditioningInfo` struct?

Reply:
@ArtsparkAI also commented in previous PR #3085 that it'd be better to avoid kwargs. 
We could possibly could put ControlNet  residuals output in ExtraConditioningInfo. But it would add two new vars in the struct, `down_block_additional_residuals` and `mid_block_additional_residuals`.  These are passed as separate args to the base model diffusers UNet2DConditionalModel.forward() call, and are handled differently from the `cross_attention_kwargs` arg. Does that still sound better than using kwargs?

Reply:
yes, definitely better than kwargs. the point of `ExtraConditioningInfo` is exactly to store this kind of stuff. wrap them in their own struct `ControlNetArgs`, add an instance of that to `ExtraConditioningInfo` that defaults to `None`.